packages/pwa/app/utils/utils.js

```diff
@@ -158,7 +160,86 @@ export const capitalize = (text) => {
 }
 
 /**
- * A util to return current url configuration
- * @returns {object} - url object from the pwa-kit.config.json file
+ * This function return the identifiers (site and locale) from the given url
+ * The site will always go before locale if both of them are presented in the pathname
+ * @param path {string}
+ * @returns {object}
  */
-export const getUrlConfig = () => getConfig().app.url
+export const getParamsFromPath = (path) => {
+    const {pathname, search} = new URL(pathToUrl(path))
+
+    const config = getConfig()
+    const {pathMatcher, searchMatcherForSite, searchMatcherForLocale} = getConfigMatcher(config)
+    const pathMatch = pathname.match(pathMatcher)
+    const searchMatchForSite = search.match(searchMatcherForSite)
+    const searchMatchForLocale = search.match(searchMatcherForLocale)
+
+    // the value can only either in the path or search query param, there will be no overridden
+    const site = pathMatch?.groups.site || searchMatchForSite?.groups.site
+
+    const locale = pathMatch?.groups.locale || searchMatchForLocale?.groups.locale
+    return {site, locale}
+}
+
+/**
+ * This function returns the url config from the current configuration
+ * @return {object} - url config
+ */
+export const getUrlConfig = () => {
+    const {app} = getConfig()
+    if (!app.url) {
+        throw new Error("Can't find any valid url config. Please check your configuration file.")
+    }
+    return app.url
+}
+
+/**
+ * This function return the regex for matching site and locale
+ * @param config
+ * @return {{searchMatcherForSite: RegExp, searchMatcherForLocale: RegExp, pathMatcher: RegExp}}
+ */
+export const getConfigMatcher = (config) => {
+    if (!config) {
+        throw new Error('Config is not defined.')
+    }
+
+    const allSites = getSites()
+    let sites = []
+    let locales = []
+    allSites.forEach((site) => {
+        sites.push(site.alias)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            locales.push(locale.id)
+        })
+    })
+
+    allSites.forEach((site) => {
+        sites.push(site.id)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            locales.push(locale.alias)
+        })
+    })
+
```

Comment:
I can't imagine that there isn't a better way of accomplishing this in one iteration. Have you considered maintaining both a `localeId` array and a `localeAlias` array and joining them in the end?

Reply:
Thanks for the suggestion. I've made the changes.